File: java/nec.unit-test/src/test/java/nec/utils/MockCountCallsEngine.java
package nec.utils;

import nec.Models.IEngine;

public class MockCountCallsEngine implements IEngine {

    private int _igniteCalls;

    private int _stopCalls;

    public void ignite() {

	this._igniteCalls++;
    }

    public void stop() {

	this._stopCalls++;
    }

    public int getIgniteCalls() {

	return this._igniteCalls;
    }

    public int getStopCalls() {

	return this._stopCalls;
    }

}

File: netCore/NEC.UnitTest/tests/NEC.Models.Tests/Utils/MockCountCallsEngine.cs
namespace NEC.Models.Tests.Utils
{
    public class MockCountCallsEngine : IEngine
    {
        public int IgniteCalls { get; private set; } = 0;

        public int StopCalls { get; private set; } = 0;

        public void Ignite() => IgniteCalls++;

        public void Stop() => StopCalls++;
    }
}
